File: schema_salad/java/main_utils/EnumLoader.java
package ${package}.utils;

import java.util.Arrays;
import java.util.List;

public class EnumLoader implements Loader<String> {
  private final List<String> symbols;

  public EnumLoader(final List<String> symbols) {
    this.symbols = symbols;
  }

  public EnumLoader(final String[] symbols) {
    this(Arrays.asList(symbols));
  }

  public String load(
      final Object doc,
      final String baseUri,
      final LoadingOptions loadingOptions,
      final String docRoot) {
    final String docString = Loader.validateOfJavaType(String.class, doc);
    if (!this.symbols.contains(docString)) {
      throw new ValidationException(String.format("Expected one of %s", this.symbols));
    }
    return docString;
  }
}

File: schema_salad/typescript/util/loaders/enumLoader.ts
import { Loader, LoadingOptions, ValidationException } from '../internal'

export class EnumLoader implements Loader {
  symbols: string[]

  constructor (symbols: string[]) {
    this.symbols = symbols
  }

  async load (doc: string, baseuri: string, loadingOptions: LoadingOptions, docRoot?: string): Promise<any> {
    if (this.symbols.includes(doc)) {
      return doc
    } else {
      throw new ValidationException(`Expected one of ${this.symbols.toString()}`)
    }
  }
}
